Switch to Vim 8 Package Manager

alexgraffeocohen opened the pull request:
This branch switches us from Pathogen to Vim 8's built-in support for managing vim plugins. This move gains us three main advantages:

1. Easy support for loading personal, local vim plugins. To do so, place any plugins int `.vim/pack/local/start`.
2. Removes the need for Pathogen, reducing the dependencies of our dotfiles
3. Requires Vim 8, meaning we can start adding features optimized for or required by that version of Vim.

I tested it on my own machine, but would love others to test out this branch to make sure it works for others. When adding local plugins, just be sure to re-source `.vimrc` when making sure they work.

Installing Vim 8 should only require running `brew install vim` and then making sure your `PATH` is configured correctly load homebrew vim first.

alexgraffeocohen committed:
Use Vim 8 plugin support instead of Pathogen

alexgraffeocohen committed:
Add path to local vim plugins to .gitignore

jnimety commented:
Look good to me. Will submodules get cleaned up automatically on homebrew sync or are there extra steps?

alexgraffeocohen commented:
@jnimety Good question that probable catches an oversight on my part! 

As a course of this work, I just copy-pasted all the folders from `.vim/bundle` to the pack folder and deleted `.vim/bundle`. When I ran `gitsubmodule status` on my branch just now, it didn't list any of the submodules. So I think moving all those folders did clear out the submodules, but I think we have to add a step to this branch to re-add those folders as git submodules, correct?

benjaminoakes commented:
_bump_

Still any interest in making this happen?

alexgraffeocohen commented:
@benjaminoakes Ah yes, I've been meaning to get back to this for a while! I hope to soon, but my schedule is pretty packed at the moment.

Commit:
move plugins from Pathogen directory to vim packages directory

```
for x in home/.vim/bundle/*
do
  git mv $x home/.vim/pack/dotfiles/start
done
```

benjaminoakes commented:
🎉 

Any special instructions for upgrading?  (Looks like just the location of the submodules changed?)

jnimety commented:
You may want to torch your existing homesick dotfiles and reinstall. `git
submodules` are a pain and don't appear to automatically move after
checkout properly.

On Tue, Jan 21, 2020 at 2:52 PM Ben Oakes <notifications@github.com> wrote:

> 🎉
>
> Any special instructions for upgrading? (Looks like just the location of
> the submodules changed?)
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/ContinuityControl/dotfiles/pull/70?email_source=notifications&email_token=AAARGVG2ZM476OPGZXJDMY3Q65N73A5CNFSM4DSIVS32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJRG3BA#issuecomment-576875908>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAARGVBKRXXVGRPS7BIMMKDQ65N73ANCNFSM4DSIVS3Q>
> .
>


-- 
Joel Nimety
Chief Infrastructure Officer
Continuity
203.672.9843 (office)
203.763.9263 (cell)
http://www.continuity.net
http://www.linkedin.com/in/jnimety

-- 


-- 
This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.
